include/zephyr/input/input.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_INPUT_H_
+#define ZEPHYR_INCLUDE_INPUT_H_
+
+/**
+ * @brief Input Interface
+ * @defgroup input_interface Input Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <stdint.h>
+#include <zephyr/device.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Input event structure.
+ *
+ * This structure represents a single input event, for example a key or button
+ * press for a single button, or an absolute or relative coordinate for a
+ * single axis.
+ */
+struct input_event {
+	/** Device generating the event or NULL. */
+	const struct device *dev;
+	/** Event type, one of INPUT_EV_*. */
+	uint16_t type;
+	/** Event code, one of INPUT_{KEY,BTN,ABS,REL,...)_*. */
```

Comment:
If using grouped definitions with anchors, you could do here:

```c
/* in ../dt-bindings/input/input-event-codes.h */
/**
 * @name Event codes
 * @anchor INPUT_EVENT_CODES
 * @{
 */
#define ....
 /** @} */

/* in ../input.h */
struct input_event {
        /** Event code (@ref INPUT_EVENT_CODES) */
        uint16_t code;
};
```

Reply:
Done, nice.